src/Pages/AdminTaskTable/components/ViewPayloadModal/ViewPayloadModal.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  Bullseye,
+  Button,
+  Modal,
+  ModalVariant,
+  Popover,
+  Spinner,
+  Tab,
+  TabContent,
+  TabTitleText,
+  Tabs,
+} from '@patternfly/react-core';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import { createUseStyles } from 'react-jss';
+import { createRef, useMemo, useState } from 'react';
+import { AdminTask, JSONTaskPayload } from '../../../../services/AdminTasks/AdminTaskApi';
+import AdminTaskInfo from './components/AdminTaskInfo';
+import ReactJson from 'react-json-view';
+
+const useStyles = createUseStyles({
+  jsonView: {
+    fontSize: '12px',
+  },
+});
+
+export interface ViewPayloadProps {
+  setClosed: () => void;
+  open: boolean;
+  isLoading: boolean;
+  adminTask: AdminTask | null;
+}
+
+export interface TabData {
+  title: string;
+  data: { [key: string]: string };
+  contentRef: React.RefObject<HTMLElement>;
+}
+
+const ViewPayloadModal = ({ adminTask, open, isLoading, setClosed }: ViewPayloadProps) => {
+  if (!open) return <></>;
+  const classes = useStyles();
+
+  const [activeTabKey, setActiveTabKey] = useState<string | number>(0);
+  const detailRef = createRef<HTMLElement>();
+
+  const handleTabClick = (_, tabIndex: string | number) => {
+    setActiveTabKey(tabIndex);
+  };
+
+  const tabs = useMemo(() => {
+    const tabs: TabData[] = [];
+
+    switch (adminTask?.typename) {
+      case 'snapshot':
+        if (adminTask.payload?.sync) {
+          tabs.push({
+            title: 'Sync',
+            data: adminTask.payload.sync,
+            contentRef: createRef<HTMLElement>(),
+          });
+        }
+        if (adminTask.payload?.distribution) {
+          tabs.push({
+            title: 'Distribution',
+            data: adminTask.payload.distribution,
+            contentRef: createRef<HTMLElement>(),
+          });
+        }
+        if (adminTask.payload?.publication) {
+          tabs.push({
+            title: 'Publication',
+            data: adminTask.payload.publication,
+            contentRef: createRef<HTMLElement>(),
+          });
+        }
+        return tabs;
+      default:
+        tabs.push({
+          title: 'Payload',
+          data: adminTask?.payload as JSONTaskPayload,
+          contentRef: createRef<HTMLElement>(),
+        });
+        return tabs;
+    }
+  }, [adminTask]);
```

Comment:
useMemo and most react hooks do not check their object dependencies deeply. 
It may make more sense for you to target specific values within adminTask for changes. 
"adminTask?.typename" and "adminTask.payload" for example may be better dependencies.

This principal should be kept in mind for most dependency based hooks. 
Basically you should strive to make your dependencies javascript "primitives".